fiftyone/utils/cvat.py

```diff
@@ -3655,6 +3657,15 @@ def _login(self, username, password):
                 "csrftoken"
             ]
 
+    @backoff.on_exception(
+        backoff.expo,
+        requests.exceptions.RequestException,
+        factor=HTTPRetryConfig.FACTOR,
+        max_tries=HTTPRetryConfig.MAX_TRIES,
+        giveup=lambda e: hasattr(e, "error_code")
```

Comment:
This logic says "never giveup if we get an exception that doesn't have an error code".

First question: why is `hasattr(e, "error_code")` needed?

Comment: the goal here is to only retry in specific cases where the server gave us a code that we know is retryable, so if we get something unexpected, that would be a giveup condition, not a retry condition

Reply:
Probably should specifically test for that exception then rather than blindly assuming all errors without codes are retryable